### DISCUSSION TITLE
Subject: Issue with Registering StarkNet Wallet Identity in IdentityKeyAPI

### DISCUSSION THREAD

**Discussion by machuwey** [WalletConnect/walletconnect-monorepo#4217] [Developer Support]
Hi,
I've been attempting to register identities using the `registerIdentity` function for both Ethereum and StarkNet wallets. The registration process works flawlessly when I use an Ethereum wallet, but it fails when I attempt to register a StarkNet wallet. Here is a simplified version of the code I'm using:

First of all, I changed the wallet generation from ETH to STARKNET:
```swift 
case .custom(let privateKey):
            let address = try! EthereumPrivateKey(hexPrivateKey: "0x" + privateKey, ctx: nil).address.hex(eip55: true)
            return Account("eip155:1:\(address)")!
```
To this
```swift 
case .custom(let privateKey):
            guard let private_felt = Felt(fromHex: privateKey) else { return Account("starknet:1:fail")! }
            let publicKey = try! StarknetCurve.getPublicKey(privateKey: private_felt)
            guard let classhash = Felt(fromHex: "0x0750a901280764b9b1d2a82d3d3ed3c1047314ec2f094e61547dd4bce4dd6671") else { return Account("starknet:1:fail")!  }
            let address = StarknetContractAddressCalculator.calculateFrom(classHash: classhash, calldata: [publicKey], salt: 0)
            return Account("starknet:1:\(address.toHex())")!
```

Here are the lines that cause the failure:
```swift
                try await Notify.instance.register(params: params, signature: signature)
```

```swift
func registerIdentity(cacao: Cacao) async throws {
    let api = IdentityKeyAPI.registerIdentity(cacao: cacao)
    try await httpService.request(service: api)
}
```

The `Cacao` object is populated with the appropriate data, which varies depending on the type of wallet being registered. Below are the differences in the `iss` field for the Ethereum and StarkNet wallets:

Ethereum Wallet:
```swift
iss: "did:pkh:eip155:1:0xe906Ee672b6C484A0948b3074660383008C34a13"
```

StarkNet Wallet:
```swift
iss: "did:pkh:starknet:1:0x3b9d7512b46495179d7f507e3934bcdee7a57a0ebcf461a3bd1a85ae37a12c3"
```

For some reason, the registration of a StarkNet wallet identity does not go through, and I suspect the issue might be related to how the `Cacao` object is handled when the `iss` field indicates a StarkNet wallet.

I would greatly appreciate any insights or suggestions  on this matter. Specifically, I'm interested in any special considerations or configurations that might be necessary for StarkNet wallet.

**Comment by machuwey** [comment]
Okey, now that I know a bit more, want I want to know is wheter is it posible to do (SWIS(Starknet)) instead of (SIWE) and what parts of the implementation I should change. I would like to know where the http registerIdentity call is going to, is it some external server? Or is it a call to a local part of the code?:
```swift
   let params = try await Notify.instance.prepareRegistration(account: importAccount.account, domain: "com.walletconnect")
                let signature = importAccount.onSign(message: params.message)
                try await Notify.instance.register(params: params, signature: signature)
```
And I guess if I want to use Starknet I should make my own crypto provider and not the `DefaultCryptoProvider()`, so maybe thats why is failing? Because right now I'm passing: 
```swift
       Web3Wallet.configure(metadata: metadata, crypto: DefaultCryptoProvider(), environment: BuildConfiguration.shared.apnsEnvironment)

        Notify.configure(
            environment: BuildConfiguration.shared.apnsEnvironment,
            crypto: DefaultCryptoProvider()
        )
```

**Comment by tris-p-kreme** [comment]
Instead of "did:pkh:starknet:1:0x3b9d7512b46495179d7f507e3934bcdee7a57a0ebcf461a3bd1a85ae37a12c3"
try "did:pkh:starknet:SN_MAIN:0x3b9d7512b46495179d7f507e3934bcdee7a57a0ebcf461a3bd1a85ae37a12c3" as you might have the wrong network id?
The one in my example is listed here in the [Starknet namespace CAIP-2 file](https://github.com/ChainAgnostic/namespaces/blob/main/starknet/caip2.md)
